Tools/jit/_stencils.py

```diff
@@ -230,35 +234,41 @@ def remove_jump(self) -> None:
         """Remove a zero-length continuation jump, if it exists."""
         hole = max(self.holes, key=lambda hole: hole.offset)
         match hole:
-            case Hole(
-                offset=offset,
-                kind="IMAGE_REL_AMD64_REL32",
-                value=HoleValue.GOT,
-                symbol="_JIT_CONTINUE",
-                addend=-4,
-            ) as hole:
+            case (
+                Hole(
+                    offset=offset,
+                    kind="IMAGE_REL_AMD64_REL32",
+                    value=HoleValue.GOT,
+                    symbol="_JIT_CONTINUE",
+                    addend=-4,
+                ) as hole
+            ):
```

Comment:
These `case` rewrites are... odd. I don't see how it's better in any way. Is there a setting of some sort that's causing this? Maybe because one case has a guard... but that doesn't seem like a good enough reason to indent every case and wrap them in parens.

Reply:
this is https://astral.sh/blog/ruff-v0.9.0#automatically-parenthesized-if-guards-in-match-case-statements (which Black doesn't do)

Reply:
This case (237-245) doesn't have an if-guard, though. Are all cases put in brackets if any have a guard?

Reply:
in that case it might just be due to the line length being shorter than it was previously? Addressing https://github.com/python/cpython/pull/133123#pullrequestreview-2812631631 might fix it

Reply:
I kept the wrap-to-80 commit separate ([`42ce014` (#133123)](https://github.com/python/cpython/pull/133123/commits/42ce0146d730001283ec193167d822f265afe18a)), and the change was before that.

Reply:
Or maybe it has to do with the name assignment to `hole`?

Reply:
Yep, it seems to be `as hole` that triggers the brackets. That does seem like a bug?

Reply:
https://github.com/astral-sh/ruff/issues/17796